src/closures.rs

```diff
@@ -91,9 +91,12 @@ fn get_inner_expr<'a>(
 ) -> &'a ast::Expr {
     if let ast::ExprKind::Block(ref block, _) = expr.kind {
         if !needs_block(block, prefix, context) {
-            // block.stmts.len() == 1
-            if let Some(expr) = stmt_expr(&block.stmts[0]) {
-                return get_inner_expr(expr, prefix, context);
+            // block.stmts.len() == 1 except with `|| {{}}`;
+            // https://github.com/rust-lang/rustfmt/issues/3844
+            if !block.stmts.is_empty() {
+                if let Some(expr) = stmt_expr(&block.stmts[0]) {
```

Comment:
Nit: recently I prefer to use `.first().map()` in a case like this to avoid array indexing as much as possible:

```rust
                if let Some(expr) = block.stmts.first().map(stmt_expr) {
```

Reply:
Makes sense, I'll update that

Reply:
Multi-line comments, very nice :raised_hands:

Reply:
Lol, I was just admiring that myself. I'm hoping to see review inline suggestions for multiline blocks soon 🤞